### PR TITLE
Add datastream fields to all datasets

### PR DESCRIPTION
**Pull request by ruflin**
It is planned to move from dataset.* to datastream.*. To make the path easier to 7.9 Elasticsearch (https://github.com/elastic/elasticsearch/pull/60592) and the Elastic Agent already ship with the datastream.* fields. Because of this, these should also be added to the mappings of the packages.

### CODE REVIEW

**Comment by ruflin** [comment]
Change for the agent can be found here: https://github.com/elastic/integrations/pull/213

**Comment by ruflin** [comment]
@mtojek I will probably need your help to get this over the line. After we get this in, what is the best way to release a new version of the packages to snapshot and then promote it through?

**Comment by mtojek** [comment]
> what is the best way to release a new version of the packages to snapshot and then promote it through?

currently it's the command `mage UpdatePackageStorage` and later on juggling packages through branches of the package-storage

File: packages/apache/dataset/access/fields/base-fields.yml [hunk 1]
```diff
@@ -1,3 +1,13 @@
+- name: datastream.type
```
Comment (mtojek) [inline packages/apache/dataset/access/fields/base-fields.yml:1]: I believe you have to update the import-beats script: https://github.com/elastic/integrations/blob/master/dev/import-beats/fields_base_fields.go

**Comment by mtojek** [comment]
I had plan to promote packages today. I can push changes in this PR too.

**Comment by ruflin** [comment]
@mtojek At the moment the Agent ships both fields so we need both mappings. But the plan is to follow up on removing `dataset.*` completely. The first step is getting rid of it in queries (dashboards), second step adjust the package spec but we plan to do this later on in small steps.

How tricky do you think it's going to be to adjust all dashboards and README's?

**Comment by ruflin** [comment]
@mtojek I'll do a follow up PR to adjust all the scripts if that is ok for you.

File: packages/apache/dataset/access/fields/base-fields.yml (hunk 1 above)
Comment (ruflin) [reply packages/apache/dataset/access/fields/base-fields.yml:1]: updated the script too.

**Comment by ruflin** [comment]
@mtojek Here is the script change PR: https://github.com/elastic/integrations/pull/216

File: dev/import-beats/fields_base_fields.go [hunk 2]
```diff
@@ -8,6 +8,22 @@ var baseFields = createBaseFields()
...
+			Name:        "datastream.dataset",
+			Type:        "constant_keyword",
+			Description: "Datastream dataset name.",
+		},
+		{
+			Name:        "datastream.namespace",
+			Type:        "constant_keyword",
+			Description: "Datastream namespace.",
+		},
+		// TODO: This should be removed as soon as it is not a requirement anymore by the validation
```
Comment (mtojek) [inline dev/import-beats/fields_base_fields.go:26]: Hmm.. shouldn't be validation in the package-registry already adjusted?

**Comment by mtojek** [comment]
You need to bump up versions in all manifest.yml files.

File: dev/import-beats/fields_base_fields.go (hunk 2 above)
Comment (ruflin) [reply dev/import-beats/fields_base_fields.go:26]: If we adjust it now, all old packages are invalid.

**Comment by ruflin** [comment]
I cherry-picked the changes from https://github.com/elastic/integrations/pull/216 into this one. I initially expected the script changes to be larger and the version bumps went into the wrong branch. Please have a look again.

File: dev/import-beats/fields_base_fields.go (hunk 2 above)
Comment (mtojek) [reply dev/import-beats/fields_base_fields.go:26]: PR for validation: https://github.com/elastic/package-registry/pull/618

once this one is pushed, you can update the reference to the commit

File: packages/mongodb/dataset/replstatus/fields/base-fields.yml [hunk 3]
```diff
@@ -1,3 +1,13 @@
+- name: datastream.type
+  type: constant_keyword
+  description: Datastream type.
+- name: datastream.dataset
+  type: constant_keyword
+  description: Datastream dataset name.
```
Comment (mtojek) [inline packages/mongodb/dataset/replstatus/fields/base-fields.yml:6]: nit: Datastream dataset

File: dev/import-beats/fields_base_fields.go (hunk 2 above)
Comment (mtojek) [reply dev/import-beats/fields_base_fields.go:26]: You can update the reference in docs.

**Comment by mtojek** [comment]
Side note:

we have some packages in the package storage: barracuda and endpoint, and there is bunch of new waiting for merge: https://github.com/elastic/package-storage/pull/181

any plan for these ones?

File: packages/mongodb/dataset/replstatus/fields/base-fields.yml (hunk 3 above)
Comment (ruflin) [reply packages/mongodb/dataset/replstatus/fields/base-fields.yml]: adjusted in all packages.

**Comment by ruflin** [comment]
My thinking for the pending packages in snapshot was: 

* Update registry first for production, so validation passes
* Later updated it for snapshot and with it also all the packages

**Comment by mtojek** [comment]
OK, I'll push first all integrations in the package-storage to the production and leave the ones that we do not own.

File: dev/import-beats/fields_base_fields.go (hunk 2 above)
Comment (ruflin) [reply dev/import-beats/fields_base_fields.go:26]: done

File: go.mod [hunk 4]
```diff
@@ -5,10 +5,12 @@ go 1.12
 require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/elastic/package-registry v0.4.1-0.20200702132954-41c150c8020e
+	github.com/elastic/elastic-package v0.0.0-20200731114746-b0437f8f05ca // indirect
```
Comment (ruflin) [inline go.mod:8]: @marcinpm I updated this on the golang side to get the most recent validation changes. It turns out that this was probably a good idea as my find / replace did not match all files.

But now I have some other strange errors and a merge conflict. Lets see what I can do.

Comment (ruflin) [reply go.mod]: Fixed with rebase and update to the script.

**Pull request merged**